### PR TITLE
[TS migration] Migrate 'WorkspaceList' page to TypeScript

### PR DESCRIPTION
**Pull request by DylanDylann**
### Fixed Issues

$ https://github.com/Expensify/App/issues/31994
PROPOSAL: NA

### Tests

- Test case 1:

1. Go to the settings screen
2. Click on "Workspaces" options to go to Workspace list screen
3. Click on "New workspace"
4. Click on the back button to go back to Workspace list screen
5. Verify that the new workspace in step 3 is displayed and all the workspaces are sorted in alphabetical order

- Test case 2:

1. Go to the settings screen
2. Click on "Workspaces" options to go to Workspace list screen
3. Click on any workspace
4. Click on three dot => delete workspace
5. Verify that the deleted workspace in step 3 disappeared

- Test case 3:

1. Sign in with account that does not have any workspace 
2. Go to the settings screen
3. Click on "Workspaces" options to go to Workspace list screen
4. Verify that the empty workspace list screen is displayed (that contains "Manage business expenses, issue cards, send invoices, and more." )

- Test case 4:

1. Open offline mode
2. Go to the settings screen
3. Click on "Workspaces" options to go to Workspace list screen
4. Click on "New workspace"
5. Click on the back button to go back to Workspace list screen
6. Verify that the new workspace in step 4 is displayed and all the workspaces are sorted in alphabetical order
7. Turn off offline mode
8. Verify that the new workspace in step 4 is displayed and all the workspaces are sorted in alphabetical order
- [x] Verify that no errors appear in the JS console

### QA Steps

- Test case 1:

1. Go to the settings screen
2. Click on "Workspaces" options to go to Workspace list screen
3. Click on "New workspace"
4. Click on the back button to go back to Workspace list screen
5. Verify that the new workspace in step 3 is displayed and all the workspaces are sorted in alphabetical order

- Test case 2:

1. Go to the settings screen
2. Click on "Workspaces" options to go to Workspace list screen
3. Click on any workspace
4. Click on three dot => delete workspace
5. Verify that the deleted workspace in step 3 disappeared

- Test case 3:

1. Sign in with account that does not have any workspace 
2. Go to the settings screen
3. Click on "Workspaces" options to go to Workspace list screen
4. Verify that the empty workspace list screen is displayed (that contains "Manage business expenses, issue cards, send invoices, and more." )

- Test case 4:

1. Open offline mode
2. Go to the settings screen
3. Click on "Workspaces" options to go to Workspace list screen
4. Click on "New workspace"
5. Click on the back button to go back to Workspace list screen
6. Verify that the new workspace in step 4 is displayed and all the workspaces are sorted in alphabetical order
7. Turn off offline mode
8. Verify that the new workspace in step 4 is displayed and all the workspaces are sorted in alphabetical order
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[Screencast from 20-01-2024 18:43:23.webm](https://github.com/Expensify/App/assets/141406735/d1e6deb1-353d-4434-8dbd-63fd90d1f99b)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[Screencast from 20-01-2024 17:59:30.webm](https://github.com/Expensify/App/assets/141406735/8fe324e0-ecde-4df1-ac7d-814c9d5a3a20)

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

[Screencast from 20-01-2024 18:02:33.webm](https://github.com/Expensify/App/assets/141406735/90252f06-2664-4452-8d5e-512e1630cee4)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

[Screencast from 17-01-2024 14:43:40.webm](https://github.com/Expensify/App/assets/141406735/2abfa2b4-c138-489a-b9d1-48aa5dc34f14)

</details>

<details>
<summary>MacOS: Desktop</summary>

[Screencast from 20-01-2024 18:10:23.webm](https://github.com/Expensify/App/assets/141406735/8e329e53-edbc-4227-ab19-2c34d1ed8925)

</details>

### CODE REVIEW

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 1]
```diff
@@ -106,28 +83,28 @@ function dismissWorkspaceError(policyID, pendingAction) {
...
 
     /**
      * Gets the menu item for each workspace
-     *
-     * @param {Object} item
-     * @param {Number} index
-     * @returns {JSX}
      */
-    function getMenuItem(item, index) {
+    const getMenuItem = (item: any, index: number) => {
```
Comment (DylanDylann) [inline src/pages/workspace/WorkspacesListPage.tsx:107]: @s77rt Do you have any suggestions about the item type here?

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 2]
```diff
@@ -154,32 +131,41 @@ function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, u
...
+    };
 
     /**
      * Add free policies (workspaces) to the list of menu items and returns the list of menu items
-     * @returns {Array} the menu item list
      */
     const workspaces = useMemo(() => {
-        const reimbursementAccountBrickRoadIndicator = !_.isEmpty(reimbursementAccount.errors) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
+        const reimbursementAccountBrickRoadIndicator = !_.isEmpty(reimbursementAccount?.errors) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
         return _.chain(policies)
```
Comment (s77rt) [inline src/pages/workspace/WorkspacesListPage.tsx:141]: Let's re-write this without lodash (I think we may have to in order to get the typescript match correctly)

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Please change the PR title to something more explanatory

**Comment by s77rt** [comment]
Can you please fix the lint / typescripts errors and let me know when this is ready for another review

**Comment by DylanDylann** [comment]
I will ping you soon

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 3]
```diff
@@ -106,28 +83,28 @@ function dismissWorkspaceError(policyID, pendingAction) {
...
 }
 
-function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, userWallet}) {
+function WorkspacesListPage({
+    policies = {},
+    allPolicyMembers = {},
+    reimbursementAccount = {},
+    userWallet = {
+        currentBalance: 0,
+    } as OnyxEntry<UserWallet>,
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx]: Confirm here https://expensify.slack.com/archives/C01GTK53T8Q/p1705434458094179

**Comment by DylanDylann** [comment]
@s77rt I`ve fixed all your comments. Please help review again

File: src/libs/PolicyUtils.ts [hunk 4]
```diff
@@ -72,12 +73,12 @@ function getUnitRateValue(customUnitRate: UnitRate, toLocaleDigit: (arg: string)
 /**
  * Get the brick road indicator status for a policy. The policy has an error status if there is a policy member error, a custom unit error or a field error.
  */
-function getPolicyBrickRoadIndicatorStatus(policy: OnyxEntry<Policy>, policyMembersCollection: OnyxCollection<PolicyMembers>): string {
+function getPolicyBrickRoadIndicatorStatus(policy: OnyxEntry<Policy>, policyMembersCollection: OnyxCollection<PolicyMembers>): ValueOf<typeof CONST.BRICK_ROAD_INDICATOR_STATUS> | undefined {
```
Comment (blazejkustra) [inline src/libs/PolicyUtils.ts:76]: Nice 👍

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 5]
```diff
@@ -1,76 +1,57 @@
-import PropTypes from 'prop-types';
+import _ from 'lodash';
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:1]: Let's remove any lodash and underscore usage if possible

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 6]
```diff
@@ -141,45 +118,55 @@ function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, u
...
                     title={keyTitle}
-                    icon={item.icon}
+                    icon={item.icon ?? ''}
                     iconType={CONST.ICON_TYPE_WORKSPACE}
                     onPress={item.action}
                     iconStyles={item.iconStyles}
                     iconFill={item.iconFill}
                     shouldShowRightIcon
-                    badgeText={getWalletBalance(isPaymentItem)}
+                    badgeText={getWalletBalance(false)}
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:127]: Why was this changed?

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 7]
```diff
@@ -106,30 +84,29 @@ function dismissWorkspaceError(policyID, pendingAction) {
...
 }
 
-function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, userWallet}) {
+function WorkspacesListPage({
+    policies = {},
+    allPolicyMembers = {},
+    reimbursementAccount = {},
+    userWallet = {
+        currentBalance: 0,
+    },
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:93]: Most likely these default values aren't needed, in Ts we have optional chaining which should handle these gracefully

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Comment by blazejkustra** [comment]
Please rename the PR to: `[TS migration] Migrate 'WorkspaceList' page to TypeScript`

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 5 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx:1]: I think we cannot remove the lodash in this file

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 6 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx:127]: Because the ```const isPaymentItem = item.translationKey === 'common.wallet';``` in this case is always false

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 7 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx]: I just removed all default values

**Comment by DylanDylann** [comment]
@blazejkustra I fixed all your comments

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 5 above)
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:1]: > Try to remove lodash and underscore usage, instead rely on native JS methods

> I think we cannot remove the lodash in this file

Why not? @DylanDylann 

- `_.isEmpty -> isEmptyObject`
- `_.chain -> native methods`
- `_.map -> native map`

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 8]
```diff
@@ -106,30 +84,22 @@ function dismissWorkspaceError(policyID, pendingAction) {
...
     /**
      * Gets the menu item for each workspace
-     *
-     * @param {Object} item
-     * @param {Number} index
-     * @returns {JSX}
      */
-    function getMenuItem(item, index) {
-        const keyTitle = item.translationKey ? translate(item.translationKey) : item.title;
-        const isPaymentItem = item.translationKey === 'common.wallet';
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:132]: Please check in runtime that translationKey isn't defined on items. I'm good to remove it but I think along with this line you should be able to remove `getWalletBalance` util, or at least refactor it.

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 5 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx]: Updated

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 8 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx:132]: I tested, the ```item.translationLey``` is always ```undefined```

Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx:132]: Also removed the ```getWalletBalance```

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 9]
```diff
@@ -141,45 +101,56 @@ function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, u
...
-                iconType: policy.avatar ? CONST.ICON_TYPE_AVATAR : CONST.ICON_TYPE_ICON,
-                action: () => Navigation.navigate(ROUTES.WORKSPACE_INITIAL.getRoute(policy.id)),
+    const workspaces: WorkspaceItem[] = useMemo(() => {
+        const reimbursementAccountBrickRoadIndicator = reimbursementAccount?.errors ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined;
+        if (isEmptyObject(policies)) {
+            return [];
+        }
+        return Object.values(policies)
+            ?.filter((policy: OnyxEntry<PolicyType>) => PolicyUtils.shouldShowPolicy(policy, !!isOffline))
+            ?.map((policy: OnyxEntry<PolicyType>) => ({
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListPage.tsx:128]:
```suggestion
        return Object.values(policies ?? {})
            .filter((policy) => PolicyUtils.shouldShowPolicy(policy, !!isOffline))
            .map((policy) => ({
```

**Review by blazejkustra — APPROVED** [review]
LGTM

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by DylanDylann** [comment]
@s77rt Thanks for your suggestions. All are updated

File: src/pages/workspace/WorkspacesListPage.tsx [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+    /**
+     * Add free policies (workspaces) to the list of menu items and returns the list of menu items
+     */
+    const workspaces: WorkspaceItem[] = useMemo(() => {
+        const reimbursementAccountBrickRoadIndicator = reimbursementAccount?.errors ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined;
+        if (isEmptyObject(policies)) {
+            return [];
+        }
+        return Object.values(policies ?? {})
```
Comment (s77rt) [inline src/pages/workspace/WorkspacesListPage.tsx:123]:
```suggestion
        return Object.values(policies)
```

Is the nullish check needed here?

Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx:123]: I think we need it because if the policies are undefined or null, Object.values(policies) will through an error

Comment (s77rt) [reply src/pages/workspace/WorkspacesListPage.tsx:123]: I'm not seeing any error. I think we already covered that case with the isEmptyObject check

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop


</details>

**Comment by s77rt** [comment]
@DylanDylann Please add the missing screenshots and write a clear testing steps

**Comment by DylanDylann** [comment]
@s77rt Added the missing screenshots and test cases. Also, I have a problem with ios native so can not test on it. Please help me on it

File: src/pages/workspace/WorkspacesListPage.tsx (hunk 10 above)
Comment (DylanDylann) [reply src/pages/workspace/WorkspacesListPage.tsx]: You are correct. Removed it

**Review by s77rt — APPROVED** [review]

**Comment by s77rt** [comment]
@francoisl

**Review by Julesssss — APPROVED** [review]
